pages/index.vue

```diff
@@ -1,63 +1,53 @@
 <template>
   <div class="expression">
-    <form @submit.prevent>
-      <label>
-        ョ米
-        <input v-model="expression.raw" type="text">
-      </label>
-    </form>
+    <h1 class="title">オンライン胡算器</h1>
+    <div class="display">
+      <input v-model="expression.raw" class="eq-input" type="text" />
+      <div class="hor-line" />
+      <div
+        :class="{
+          result: true,
+          'error-result': expression.isResultError,
+        }"
+      >
+        {{ expression.calc(expression.raw) }}
+      </div>
+    </div>
     <ul class="input_buttons">
       <li>
-        <button class="ui_button" @click="expression.raw += '7'">
-          7
-        </button>
+        <button class="ui_button" @click="expression.raw += '7'">7</button>
```

Comment:
ここですが,DRY規則の気持ちから
```suggestion
        <button class="ui_button" @click="expression.append">7</button>
```
のようにインラインハンドラからメソッドハンドラにして

```typescript
append: function(event){
  this.raw += event.name
}
```
をexpressionのメソッドに定義してあげたらスッキリすると思いました.

Reply:
なるほど〜

やろうとしてみたんですが、TS Vueで `e.target` からHTMLのinner textを取得するのが意外とむずいっぽく、直接書いちゃうのが一番簡潔かもみたいになってます

代わりに該当箇所で、list renderで別のスッキリを生んだのですがこれはどうでしょうか！